app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  before_save{self.email = email.downcase}
+  validates :name, presence: true, length: {maximum: Setting.User.name.maximum}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true,
+    length: {maximum: Setting.User.email.maximum},
+    format: {with: VALID_EMAIL_REGEX}, uniqueness: {case_sensitive: false}
```

Comment:
giải thích dòng 5,6,7

Reply:
Trường validates là email, giá trị email được coi là hợp lệ khi nó thỏa mãn 4 điều sau:
- không được để trống
- độ dài ký tự không vượt quá giá trị đã được định nghĩa trong config 
- giá trị phải chuẩn theo mẫu được định nghĩa trong hằng  VALID_EMAIL_REGEX
- giá trị phải là duy nhất (chưa tồn tại trong csdl ) , dù khác nhau trong cách viết in hoa hay in thường cũng không được chấp nhận

Reply:
good